test/run_test.py

```diff
@@ -47,22 +50,20 @@
 }
 
 
+def print_to_stderr(message):
+    # Print to stderr because test output also goes to stderr. This ensures
+    # synchronization between the two output sources.
+    print(message, file=sys.stderr)
+
+
 def shell(command, cwd):
-    popen = subprocess.Popen(
-        command,
-        stdout=subprocess.PIPE,
-        universal_newlines=True,
-        cwd=cwd,
-        shell=True)
-    for stdout_line in iter(popen.stdout.readline, ''):
-        print(stdout_line.strip('\n'))
-    popen.stdout.close()
-    return_code = popen.wait()
-    return return_code == 0
+    # bufsize = 1 means line-buffered output (rather than fully-buffered)
+    return subprocess.call(
+        shlex.split(command), universal_newlines=True, bufsize=1, cwd=cwd) == 0
```

Comment:
Is `shell` incompatible with `bufsize`? What's the benefit in removing it?

Reply:
I'm actually not sure if this works fine with default buffering setting, I'll try.